src/engine/Server.cpp

```diff
@@ -164,6 +164,27 @@ void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
   httpServer.run();
 }
 
+std::optional<std::string> Server::extractAccessToken(
+    const ad_utility::httpUtils::HttpRequest auto& request,
+    const ad_utility::url_parser::ParamValueMap& params) {
+  if (request.find(http::field::authorization) != request.end()) {
+    string_view authorization = request[http::field::authorization];
+    const std::string prefix = "Bearer ";
+    if (!authorization.starts_with(prefix)) {
+      throw std::runtime_error(absl::StrCat(
+          "Authorization header must start with \"Bearer \". Got: \"",
+          authorization, "\"."));
+    }
+    authorization.remove_prefix(prefix.length());
+    return std::string(authorization);
+  }
+  if (params.contains("access-token")) {
+    return ad_utility::url_parser::getParameterCheckAtMostOnce(params,
+                                                               "access-token");
+  }
+  return std::nullopt;
```

Comment:
I think we shouldn't have a silent precedence.
What about "If both are specified, then they must be equal (the redundancy is okay, maybe someone has redundant APIs", but two different tokens trigger and exception?